miio/waterheater.py

```diff
@@ -0,0 +1,219 @@
+import enum
+import logging
+from typing import Any, Dict
+
+import click
+
+from .click_common import EnumType, command, format_output
+from .device import Device, DeviceStatus
+from .exceptions import DeviceException
+
+_LOGGER = logging.getLogger(__name__)
+
+# VIOMI Internet electric water heater 1A (60L)
+MODEL_WATERHEATER_E1 = "viomi.waterheater.e1"
+AVAILABLE_PROPERTIES_E1 = [
+    "washStatus",
+    "velocity",
+    "waterTemp",
+    "targetTemp",
+    "errStatus",
+    "hotWater",
+    "needClean",
+    "modeType",
+    "appointStart",
+    "appointEnd",
+]
+
+SUPPORTED_MODELS: Dict[str, Dict[str, Any]] = {
+    MODEL_WATERHEATER_E1: {
+        "temperature_range": (30, 75),
+        "bacteriostatic_mode": True,
+        "bacteriostatic_temperature": 80,
+        "available_properties": AVAILABLE_PROPERTIES_E1,
+    },
+}
+
+class OperationMode(enum.Enum):
+    Termostatic = 0
+    Heating = 1
+    Booking = 2
+ 
+class OperationStatus(enum.Enum):
+    Off = 0
+    Heating = 1
+    KeepWarm = 2
+
+class WaterHeaterException(DeviceException):
+    pass
+
+class WaterHeaterStatus (DeviceStatus):
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """Response of a Waterheater (viomi.waterheater.e1):
+        {'washStatus': 1, 'velocity': 0, 'waterTemp': 29,
+        'targetTemp': 70, 'errStatus': 0, 'hotWater': 60,
+        'needClean': 0, 'modeType': 1, 'appointStart': 7
+        'appointEnd': 0}
+        """
+        self.data = data
+
+    @property
+    def status(self) -> OperationStatus:
+        return OperationStatus(self.data["washStatus"])
+
+    @property
+    def is_on(self) -> bool:
+        """True if device is currently on."""
+        return not self.status == OperationStatus.TurnedOff
+
+    @property
+    def velocity(self) -> int:
+        return self.data["velocity"]
+
+    @property
+    def water_temperature(self) -> int:
+        """Current water temperature."""
+        return self.data["waterTemp"]
+
+    @property
+    def target_temperature(self) -> int:
+        """Target water temperature."""
+        return self.data["targetTemp"]
+
+    @property
+    def error(self) -> int:
+        return self.data["errStatus"]
+
+    @property
+    def hot_water_volume(self) -> int:
+        return self.data["hotWater"]
+
+    @property
+    def cleaning_required(self) -> int:
+        return self.data["needClean"]
+
+    @property
+    def mode(self) -> OperationMode:
+        return OperationMode(self.data["modeType"])
+
+    @property
+    def booking_time_start(self) -> int:
+        return self.data["appointStart"]
+
+    @property
+    def booking_time_end(self) -> int:
+        return self.data["appointEnd"]
```

Comment:
What is the meaning of these? What do the values (7 and 0) in the example response mean?